src/main/java/com/fasterxml/jackson/databind/deser/std/StdDeserializer.java

```diff
@@ -1421,13 +1421,16 @@ protected final String _parseString(JsonParser p, DeserializationContext ctxt,
             return ob.toString();
         // 29-Jun-2020, tatu: New! "Scalar from Object" (mostly for XML)
         case JsonTokenId.ID_START_OBJECT:
+            act = _checkObjectToStringCoercion(p, ctxt, rawTargetType);
             return ctxt.extractScalarFromObject(p, this, _valueClass);
```

Comment:
... here I think we do need to pass `rawTargetType` and NOT `_valueClass`. And that should do the trick.